### PR TITLE
MohammadH/ Fix Academy TS errors

### PR DESCRIPTION
**Pull request by mohammad-rework**
Changes:

-   Academy TS errors and code smells are resolved in this PR

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/elements/background-image.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const Background = ({
 
     return (
         <StyledBackground Tag="div" style={style} $dark={dark} {...bg_image} {...props}>
-            {children}
+            {children ? children : <></>}
```
Comment (niloo-fs) [inline src/components/elements/background-image.tsx:47]: Can we write this like this? 
`{children && children}`

File: src/pages/academy/components/search_banner/_search-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        0% {
+            opacity: 0;
+        }
+        100% {
+            opacity: 1;
+        }
+    }
+`
+const TopicParent = styled(Flex)`
+    display: ${(props) => (props.modal ? 'flex' : 'none')};
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-banner.tsx:32]: Can we destructure props like this: `({ modal }) => (modal ? 'flex' : 'none')`

File: src/pages/academy/components/search_banner/_search-banner.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+const SearchBanner = ({ hidden }: SearchBannerProps) => {
+    const [is_mobile] = useBrowserResize(768)
+    const [modal_opened, setModal] = useState(false)
+    const [hide_mobile_topic, setHideMobileTopic] = useState(false)
+    const [blog_post_url, setBlogPostURL] = useState(false)
+    const { is_eu, is_uk } = useCountryRule()
+
+    // Filter out restricted categories from the combined filter type array based on geolocation
+    useEffect(() => {
+        combined_filter_type.forEach((type) => {
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-banner.tsx:82]: Same for here 
`
type.items = type.items.filter(({ is_visible_eu, is_visible_uk }) => {
                if (is_eu) return is_visible_eu
                if (is_uk) return is_visible_uk
                return obj
            })
`

File: src/pages/academy/components/search_banner/_search-nav.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { Flex } from 'components/containers'
+import device from 'themes/device'
+import { Header } from 'components/elements'
+import AcademyLogo from 'images/svg/blog/academy-text.svg'
+
+type NavProps = {
+    setModal?: React.Dispatch<React.SetStateAction<boolean>>
+    openModal: () => void
+    Chevron: string
+    modal_opened: boolean
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-nav.tsx:15]: We have a role in the deriv-app project: Those variables which store a boolean value, should start with `is_`, `has_`, ...
I don't know we have it in deriv.com or not but base on this role we can change this to `is_modal_opened`

File: src/pages/academy/components/search_banner/_search-banner.tsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    max-width: 250px;
+    padding: 16px;
+    height: auto;
+
+    @media ${device.laptopM} {
+        max-width: 50%;
+    }
+
+    @media ${device.tabletL} {
+        display: ${(props) => (props.is_mobile_expanded ? 'none' : 'flex')};
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-banner.tsx:65]: I guess you need to define a types for styles props as well

File: src/components/elements/background-image.tsx (hunk 1 above)
Comment (mohammad-rework) [reply src/components/elements/background-image.tsx:47]: I've tried this and I got this error:
error

File: src/pages/academy/components/search_banner/_search-banner.tsx (hunk 3 above)
Comment (mohammad-rework) [reply src/pages/academy/components/search_banner/_search-banner.tsx:82]: If I do it I cannot use `return obj` in last line.

File: src/pages/academy/components/search_banner/_search-nav.tsx (hunk 4 above)
Comment (prince-deriv) [reply src/pages/academy/components/search_banner/_search-nav.tsx:15]: @niloo-fs  that convention is also applied in deriv.com.

File: src/components/elements/background-image.tsx (hunk 1 above)
Comment (niloo-fs) [reply src/components/elements/background-image.tsx:47]: alrighty

File: src/pages/academy/components/search_banner/_search-nav.tsx (hunk 4 above)
Comment (niloo-fs) [reply src/pages/academy/components/search_banner/_search-nav.tsx]: Thank u @prince-deriv

File: src/pages/academy/components/search_banner/_search-banner.tsx (hunk 3 above)
Comment (niloo-fs) [reply src/pages/academy/components/search_banner/_search-banner.tsx:86]: Alrighty

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+import { DerivStore } from 'store'
+import { useDebouncedEffect } from 'components/hooks/use-debounced-effect'
+import device from 'themes/device'
+import { Flex } from 'components/containers'
+import { useBrowserResize } from 'components/hooks/use-browser-resize'
+import { slugify } from 'common/utility'
+import { Header } from 'components/elements'
+
+type ElementWithMaximiseProps = {
+    maximise?: boolean
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-bar.tsx:27]: is_maximise

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+const FlexSearchBar = styled(Flex)`
+    max-width: 400px;
+    width: 100%;
+    background-color: rgba(236, 241, 247, 0.5);
+    border: 2px solid #ecf1f7;
+    border-radius: 20px;
+    position: relative;
+    overflow: hidden;
+    transition: max-width 500ms ease;
+    ${(props) =>
```
Comment (niloo-fs) [inline src/pages/academy/components/search_banner/_search-bar.tsx:45]: you can define types for here

File: src/pages/academy/components/video-banner/carousel/_custom-carousel.tsx [hunk 8]
```diff
@@ -251,7 +277,7 @@ export const CustomCarousel = ({
                 {/*  TODO: will remove later,not using this for now */}
                 {nav_color && (
                     <NavigationContainer bottom_offset={bottom_offset}>
-                        {children.map((child, idx) => (
+                        {children.map((_child, idx) => (
```
[the remark this answers is not in the document]
Comment (mohammad-rework) [reply src/pages/academy/components/video-banner/carousel/_custom-carousel.tsx:280]: I couldn't find any specific key to pass via children :cry:

File: src/components/layout/nav/components/nav-template.tsx [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import React, { ReactNode } from 'react'
 import styled from 'styled-components'
-import SearchBanner from 'pages/academy/components/_search-banner'
+import SearchBanner from 'pages/academy/components/search_banner/_search-banner'
```
Comment (prince-deriv) [inline src/components/layout/nav/components/nav-template.tsx:3]: Can we have an `index.ts` inside components so we export everything from there?

**Review by niloo-fs — APPROVED** [review]
LGTM

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 10]
```diff
@@ -0,0 +1,419 @@
...
+                            <Line />
+                            <SearchResultRows
+                                jc="flex-start"
+                                ai="center"
+                                is_active={focus_index === -1}
+                                style={{ color: 'var(--color-blue-3)' }}
+                                onMouseDown={handleSearchClick}
+                            >{`Search for: ${search_query}`}</SearchResultRows>
+
+                            {data_to_render &&
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_search-bar.tsx:329]: what do you think about this? 
```suggestion
                            {data_to_render?.slice(0, 6)?.map(...)
```
I think it's gonna be cleaner this way, but if you don't like it ignore this comment. :pray:

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+
+                                    return (
+                                        <SearchResultRows
+                                            key={post.blog_title || post.video_title}
+                                            jc="flex-start"
+                                            ai="center"
+                                            onMouseDown={handleMouseDown}
+                                            is_active={focus_index === idx}
+                                        >
+                                            {
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_search-bar.tsx:346]: do we need this `{}` here?

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+            </FormWrapper>
+            <MobileWrapper>
+                <SearchSuggestionWrapper max_width="100%" fd="column">
+                    {search_query && (
+                        <SearchResultRows
+                            jc="flex-start"
+                            ai="center"
+                            is_active={focus_index === -1}
+                            style={{ color: 'var(--color-blue-3)' }}
+                        >{`Search for: ${search_query}`}</SearchResultRows>
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_search-bar.tsx:375]: don't we need translation for this part?

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+                    )}
+                    {data_to_render &&
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_search-bar.tsx:377]: the same suggestion as above, :pray:

File: src/pages/academy/components/search_banner/_search-bar.tsx [hunk 14]
```diff
@@ -0,0 +1,419 @@
...
+                                <SearchResultRows
+                                    key={post.blog_title || post.video_title}
+                                    jc="flex-start"
+                                    ai="center"
+                                    onMouseDown={handleMouseDown}
+                                    is_active={focus_index === idx}
+                                    tabletL={{ ai: 'flex-start' }}
+                                >
+                                    {
+                                        <>
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_search-bar.tsx:396]: do we need this `{}` here?

File: src/pages/academy/components/search_banner/_topic-items-accordion.tsx [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+    :nth-last-child(-n + 2) {
+        max-height: ${({ is_expanded }) => (is_expanded ? '110px' : '0')};
+    }
+`
+
+const TopicItemsAccordion = ({ items, setModal, handleHref }: TopicItemsAccordionProps) => {
+    const [is_expanded, setExpanded] = useState(false)
+
+    const toggleExpand = () => {
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_topic-items-accordion.tsx:29]:
```suggestion
    const toggleExpand = () => {
        setExpanded(prevState => !prevState)
    }
```
just a suggestion, see if you like it or not, you can just ignore this comment :pray:
explanation: 
the main difference is, this way we are not using the state inside the function, normally we have to wrap it inside `useCallback` but this way you don't have to do it.

File: src/pages/academy/components/search_banner/_topic-items-accordion.tsx [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+        setExpanded(!is_expanded)
+    }
+
+    const handleModal = () => {
```
Comment (michio-rework) [inline src/pages/academy/components/search_banner/_topic-items-accordion.tsx:33]:
```suggestion
    const closeModal = () => { ... }
```
the name seems better with `closeModal`, what do you think? 
you can ignore this :pray:

File: src/pages/academy/components/search_banner/_search-bar.tsx (hunk 12 above)
Comment (mohammad-rework) [reply src/pages/academy/components/search_banner/_search-bar.tsx:375]: We only support EN in academy.

**Review by michio-rework — APPROVED** [review]

**Pull request merged**